src/MagnumPlugins/KtxImporter/KtxImporter.cpp

```diff
@@ -0,0 +1,891 @@
+
+    /* Read data format descriptor (DFD) */
+    {
+        /* Only do some very basic sanity checks, the DFD is terribly
+           over-engineered and the data is redundant if we have a
+           (Compressed)PixelFormat */
+        bool valid = false;
+        const auto descriptorData = f->in.suffix(header.dfdByteOffset).prefix(header.dfdByteLength);
+        const UnsignedInt length = *reinterpret_cast<UnsignedInt*>(descriptorData.data());
+        Utility::Endianness::littleEndianInPlace(length);
+        if(length == descriptorData.size()) {
+            auto& block = *reinterpret_cast<Implementation::KdfBasicBlockHeader*>(descriptorData.suffix(sizeof(length)).data());
+            Utility::Endianness::littleEndianInPlace(block.vendorId,
+                block.descriptorType, block.versionNumber,
+                block.descriptorBlockSize);
+
+            /* Basic block must be the first block in the DFD */
+            if(block.vendorId == Implementation::KdfBasicBlockHeader::VendorId::Khronos &&
+                block.descriptorType == Implementation::KdfBasicBlockHeader::DescriptorType::Basic &&
+                block.versionNumber == Implementation::KdfBasicBlockHeader::VersionNumber::Kdf1_3 &&
+                block.descriptorBlockSize > sizeof(block) &&
+                block.descriptorBlockSize + sizeof(length) <= length)
+            {
+                valid = true;
+
+                /* Check if pixel/block size and channel count match the format */
+                if(f->pixelFormat.isCompressed) {
+                    /* Block size */
+                    const Vector4i expected = Vector4i::pad(compressedBlockSize(f->pixelFormat.compressed), 1);
+                    const Vector4i actual{Math::max(Vector4ub::from(block.texelBlockDimension), UnsignedByte(1))};
+                    valid = valid && actual == expected;
+                } else {
+                    /* Pixel size. For supercompressed data, bytePlanes is all
+                       zeros to indicate an unsized format. */
+                    /** @todo Does this work with depth-stencil formats? */
+                    if(header.supercompressionScheme == Implementation::SuperCompressionScheme::None) {
+                        const UnsignedInt expected = f->pixelFormat.size;
+                        const UnsignedInt actual = block.bytesPlane[0];
+                        valid = valid && actual == expected;
+                    }
+                    /* Channel count */
+                    const UnsignedInt expected = f->pixelFormat.size / f->pixelFormat.typeSize;
+                    const UnsignedInt actual = (block.descriptorBlockSize - sizeof(block))/sizeof(Implementation::KdfBasicBlockSample);
+                    valid = valid && actual == expected;
+                }
+            }
+        }
+
+        if(!valid) {
+            Error{} << "Trade::KtxImporter::openData(): invalid data format descriptor";
+            return;
+        }
```

Comment:
I'm OK with dropping it, less code to test 😄 Mismatching `vkFormat`/DFD is simply a violation of the spec but there really isn't much we can do, and the `vkFormat` is our ground truth anyway. We *will* need rudimentary DFD parsing if we implement Basis support since vkFormat will be `UNDEFINED`.

Reply:
For Basis I'd only detect it and then delegate the whole thing to `BasisImporter` (opening the data there), similarly as `AnyImageImporter` does. I think that shouldn't need any messing with DFD.

But in a followup step, let's keep this PR in a reasonable scope :)